pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -619,9 +638,16 @@ func (p *PriorityQueue) Activate(logger klog.Logger, pods map[string]*v1.Pod) {
 	}
 }
 
+func (p *PriorityQueue) existsInActiveQ(pInfo *framework.QueuedPodInfo) bool {
+	p.activeQLock.RLock()
+	defer p.activeQLock.RUnlock()
+	_, exists, _ := p.activeQ.Get(pInfo)
+	return exists
+}
+
 func (p *PriorityQueue) activate(logger klog.Logger, pod *v1.Pod) bool {
 	// Verify if the pod is present in activeQ.
-	if _, exists, _ := p.activeQ.Get(newQueuedPodInfoForLookup(pod)); exists {
+	if p.existsInActiveQ(newQueuedPodInfoForLookup(pod)) {
```

Comment:
I think it won't make much difference in performance, so I would keep it

Reply:
Actually, can we remove `activate` fn, and use `moveToActiveQ` in `Activate`?
Maybe it'd require a bit of change in the parameter of `moveToActiveQ` though.

This can be follow-up anyway.